### PR TITLE
COMMERCE-12579

### PR DESCRIPTION
**Pull request by jaydawu**
https://liferay.atlassian.net/browse/COMMERCE-12579

Resent from https://github.com/liferay-commerce/liferay-portal/pull/4353

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 1]
```diff
@@ -184,6 +184,109 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12579. As a buyer, I can view the discontinued SKU shown in the product's detail replacement tab"
+	@priority = 3
+	test CanReplacementSkusShownInProductDetails {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:189]: CanViewDiscontinuedReplacementSkusInProductDetails

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 2]
```diff
@@ -184,6 +184,109 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12579. As a buyer, I can view the discontinued SKU shown in the product's detail replacement tab"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:187]: As a buyer, I can view discontinued SKUs shown in the replacement tab of the product's details page.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 3]
```diff
@@ -184,6 +184,109 @@ definition {
...
+
+				SelectFrame(locator1 = "IFrame#IFRAME");
+
+				CommerceProducts.discontinueSKU();
+
+				CommerceProducts.setReplacementSKU(replacementSKU = "MIN93015");
+			}
+		}
+
+		task ("When the buyer navigates to ABS Sensor's detail page") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:213]: When the buyer navigates to ABS Sensor's product details page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 4]
```diff
@@ -184,6 +184,109 @@ definition {
...
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/p/abs-sensor");
+		}
+
+		task ("Then assert that the Replacements button shown") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:224]: Is there a button or tab for replacement products?

"Then can assert that the Replacements tab is shown"

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 5]
```diff
@@ -184,6 +184,109 @@ definition {
...
+			for (var sku : list "MIN93016A,MIN93016B,MIN93016C,MIN93027,MIN93021") {
+				LexiconEntry.viewEntryName(rowEntry = ${sku});
+			}
+		}
+
+		task ("When the pagination is changed to 4") {
+			Pagination.changePagination(itemsPerPage = "4 Items");
+		}
+
+		task ("Then the five products are divided into two pages to display with pagination results") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:251]: Then the pagination results are shown accordingly

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 6]
```diff
@@ -184,6 +184,109 @@ definition {
...
+			Click(locator1 = "Pagination#NEXT_LINK");
+
+			Pagination.viewResults(results = "Showing 5 to 5 of 5 entries.");
+
+			AssertElementPresent(
+				key_entryName = "MIN93021",
+				locator1 = "CommerceNavigation#ENTRY_NAME");
+		}
+
+		task ("When a product Wear Sensors is searched on the page") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:267]: When the product 'Wear Sensors' is searched on the page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase [hunk 7]
```diff
@@ -184,6 +184,109 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12579. As a buyer, I can view the discontinued SKU shown in the product's detail replacement tab"
+	@priority = 3
+	test CanReplacementSkusShownInProductDetails {
+		property portal.acceptance = "false";
+
+		task ("Given 5 products are set as Discontinued with a same replacement product ABS Sensor") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase:192]: Is it possible to use this JSON macro here? 

CommerceJSONProductsAPI._patchCommerceSku();

**Review by baharturk — CHANGES_REQUESTED** [review]
Please take a look at my comments. Thanks!

**Comment by jaydawu** [comment]
ci:stop

**Comment by jaydawu** [comment]
ci:test:sf

**Review by baharturk — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
Hi [@victoroscuro](https://github.com/victoroscuro), there are no related issues to this pull, please help force-forward it, thanks!

**Comment by victoroscuro** [comment]
ci:forward:force

**Pull request closed without merging**